src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -0,0 +1,2577 @@
+        int j;
+    }
+}
+public class C1
+{
+    public static C1 operator ++(C1 obj)
+    {
+        return obj;
+    }
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopNewInInitializer2_MissingBothParens()
+        {
+            // only adding one closing paren
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        for (C1 i = new C1($$
+        int j;
+    }
+}
+public class C1
+{
+    public static C1 operator ++(C1 obj)
+    {
+        return obj;
+    }
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopInitializer()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""$$) i < 10; i++)
+";
+
+            var expected =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd"");$$ i < 10; i++)
+";
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopInitializer_MissingParen()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""$$ i < 10; i++)
+";
+
+            var expected =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd"";$$ i < 10; i++)
+";
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopCondition()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < s.IndexOf(""x""$$) i++)
+";
+
+            var expected =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < s.IndexOf(""x"");$$ i++)
+";
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopIncrement()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < s.IndexOf(""x""); i = i.IndexOf(""x""$$))
+";
+
+            var expected =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < s.IndexOf(""x""); i = i.IndexOf(""x"");$$)
```

Comment:
>;$$ [](start = 83, length = 3)

Unlike the two tests above, this one is actually introducing a syntax error. Should we instead bail out for this case?

Reply:
Yes, I fixed this.